docs/core/entity/climate.md

```diff
@@ -139,6 +139,48 @@ class MyClimateEntity(ClimateEntity):
         """Set new target hvac mode."""
 ```
 
+### Turn on
+
+```python
+class MyClimateEntity(ClimateEntity):
+    # Implement one of these methods if the device supports it.
+    # Make sure that hvac_mode is properly adjusted as well.
+
+    def turn_on(self):
+        """Turn the entity on."""
+
+    async def async_turn_on(self):
+        """Turn the entity on."""
+```
+
+### Turn off
+
```

Comment:
We should explain that the expected outcome of a successful `climate.turn_off` call is that the hvac_mode is `HVACMode.OFF`, either optimistically set by the service handler or with the next state update.